src/index.ts

```diff
@@ -1,923 +1,1064 @@
-import path				from 'path';
-import logger				from '@whi/stdlog';
-
-const log				= logger(path.basename( __filename ), {
-    level: process.env.LOG_LEVEL || 'fatal',
+import path from 'path';
+import fs from 'fs';
+import logger from '@whi/stdlog';
+import { sprintf } from 'sprintf-js';
+import crypto from 'crypto';
+import http from 'http';
+import concat_stream from 'concat-stream';
+import SerializeJSON from 'json-stable-stringify';
+import { Codec } from '@holo-host/cryptolib';
+import { Package } from '@holo-host/data-translator';
+import { HcAdminWebSocket, HcAppWebSocket } from "../websocket-wrappers/holochain/client";
+import { Server as WebSocketServer } from './wss';
+
+const log = logger(path.basename(__filename), {
+  level: process.env.LOG_LEVEL || 'fatal',
 });
 
-import crypto				from 'crypto';
-import http				from 'http';
-import concat_stream			from 'concat-stream';
-import SerializeJSON			from 'json-stable-stringify';
-import { Codec }			from '@holo-host/cryptolib';
-import { sprintf }			from 'sprintf-js';
-import { Server as WebSocketServer,
-	 Client as WebSocket }		from './wss';
-import mocks				from './mocks';
-
-
-const sha256				= (buf) => crypto.createHash('sha256').update( Buffer.from(buf) ).digest();
-const digest				= (data) => Codec.Digest.encode( sha256( typeof data === "string" ? data : SerializeJSON( data ) ));
-
-const WS_SERVER_PORT			= 4656; // holo
-const WH_SERVER_PORT			= 9676; // worm
-const RPC_CLIENT_OPTS			= {
-    "reconnect_interval": 1000,
-    "max_reconnects": 300,
+// NB: The hha installed app id must match the id provided in the hpos config when installing hha as an internal host app
+const HHA_INSTALLED_APP_ID = 'holo-hosting-app';
+
+const sha256 = (buf) => crypto.createHash('sha256').update(Buffer.from(buf)).digest();
+const digest = (data) => Codec.Digest.encode(sha256(typeof data === "string" ? data : SerializeJSON(data)));
+
+const WS_SERVER_PORT = 4656; // holo
+const WH_SERVER_PORT = 9676; // worm
+const RPC_CLIENT_OPTS = {
+  "reconnect_interval": 1000,
+  "max_reconnects": 300,
 };
-const CONDUCTOR_TIMEOUT			= RPC_CLIENT_OPTS.reconnect_interval * RPC_CLIENT_OPTS.max_reconnects;
-const NAMESPACE				= "/hosting/";
-const READY_STATES			= ['CONNECTING', 'OPEN', 'CLOSING', 'CLOSED'];
+const CONDUCTOR_TIMEOUT = RPC_CLIENT_OPTS.reconnect_interval * RPC_CLIENT_OPTS.max_reconnects;
+const NAMESPACE = "/hosting/";
+const READY_STATES = ['CONNECTING', 'OPEN', 'CLOSING', 'CLOSED'];
 
 interface CallSpec {
-    instance_id?	: string;
-    zome?		: string;
-    function?		: string;
-    args?		: any;
+  hha_hash: string;
```

Comment:
Inquiry: Where do we use CallSpec? and why would we only want to pass just the hha_hash and dna_alias?

Reply:
CallSpec is the call signature for the holo/call from chaperone.   `hha_hash` and `dna_alias` are the values it tracks and we need for either our servicelogger call or conductor calls.